11_Robots/ghost_tank/src/tank_robot_plugin.cpp

```diff
@@ -390,23 +432,62 @@ void TankRobotPlugin::teleop(double current_time)
 //   }
 // }
 
-void TankRobotPlugin::worldOdometryUpdateCallback(const nav_msgs::msg::Odometry::SharedPtr msg)
+void TankRobotPlugin::imuUpdateCallback(const sensor_msgs::msg::Imu::SharedPtr msg)
 {
-  double theta = ghost_util::quaternionToYawRad(
-    msg->pose.pose.orientation.w,
-    msg->pose.pose.orientation.x,
-    msg->pose.pose.orientation.y,
-    msg->pose.pose.orientation.z);
-  m_tank_model_ptr->setWorldPose(msg->pose.pose.position.x, msg->pose.pose.position.y, theta);
-  m_tank_model_ptr->setWorldTwist(
-    msg->twist.twist.linear.x,
-    msg->twist.twist.linear.y,
-    msg->twist.twist.angular.z);
+  // TODO do we even need this anymore
+  m_imu_yaw = ghost_util::quaternionToYawRad(
+    msg->orientation.w,
+    msg->orientation.x,
+    msg->orientation.y,
+    msg->orientation.z);
 }
 
+//void TankRobotPlugin::worldOdometryUpdateCallback(const nav_msgs::msg::Odometry::SharedPtr msg)
+//{
+//if (!m_use_backup_estimator) {
+//  double theta = ghost_util::quaternionToYawRad(
+//    msg->pose.pose.orientation.w,
+//    msg->pose.pose.orientation.x,
+//    msg->pose.pose.orientation.y,
+//    msg->pose.pose.orientation.z);
+//  m_tank_model_ptr->setWorldPose(msg->pose.pose.position.x, msg->pose.pose.position.y, theta);
+//  m_tank_model_ptr->setWorldTwist(
+//    msg->twist.twist.linear.x,
+//    msg->twist.twist.linear.y,
+//    msg->twist.twist.angular.z);
+//}
+//}
+
+//void TankRobotPlugin::worldOdometryUpdateCallbackBackup(
+//  const nav_msgs::msg::Odometry::SharedPtr msg)
+//{
+//if (m_use_backup_estimator) {
+//  double theta = ghost_util::quaternionToYawRad(
+//    msg->pose.pose.orientation.w,
+//    msg->pose.pose.orientation.x,
+//    msg->pose.pose.orientation.y,
+//    msg->pose.pose.orientation.z);
+//  m_tank_model_ptr->setWorldPose(msg->pose.pose.position.x, msg->pose.pose.position.y, theta);
+//  m_tank_model_ptr->setWorldTwist(
+//    msg->twist.twist.linear.x,
+//    msg->twist.twist.linear.y,
+//    msg->twist.twist.angular.z);
+//}
+//}
```

Comment:
i have no clue what to do with odom callback stuff, is that post particle filter??